tuning/data/setup_dataprocessor.py

```diff
@@ -491,6 +495,72 @@ def dump_dataset(
         raise RuntimeError(f"Failed to dump dataset due to error {e}") from e
 
 
+def process_dataargs_odm(
```

Comment:
nit: could we call this `setup_train_dataset_for_odm` 

because you are not touching eval_dataset

Reply:
True, done!